build/Band_Representations.jl

```diff
@@ -0,0 +1,310 @@
+end
+
+
+function get_maximal_lgs(sgnum::Integer, D::Integer)
+    maximal_lgs=Dict{String, LittleGroup{D}}()
+    for (klab, lg) in get_littlegroups(sgnum, D)
+        isspecial(kvec(lg)) || continue
+            push!(maximal_lgs, klab => lg)
+    end
+    return maximal_lgs
+end
+
+
+function make_bandrep_set(sgnum::Integer; D::Integer=2)
+    #klab2kv = Dict(klab => kvec(lg) for (klab, lg) in get_littlegroups(sgnum, D))
+    klab2kv = Dict(klab => kvec(lg) for (klab, lg) in get_maximal_lgs(sgnum, D))
+
+    irvec2d=return_irvec2d(sgnum, 2)
+    klabs=Array{String, 1}(undef, 0)
+    kvs=Array{KVec{D}, 1}(undef, 0)
+    for (klab, kv) in klab2kv
+        push!(klabs, klab)
+        push!(kvs, kv)
+       # if isspecial(kv)==true 
+      #  end
+    end
+
+    spinful, allpaths, timeinvar=false, false, true
+    
+    #irlabs=[([[lgg.cdml for lgg in lg] for (klab, lg) in get_lgirreps(sgnum, D)]...)...]
+    #irdims=[([[size(lgg.matrices[1])[1] for lgg in lg] for (klab, lg) in get_lgirreps(sgnum, D)]...)...]
+
+    irlabs=[([[lgg.cdml for lgg in lg] for (klab, lg) in get_maximal_lgirreps(sgnum, D)]...)...]
+    irdims=[([[size(lgg.matrices[1])[1] for lgg in lg] for (klab, lg) in get_maximal_lgirreps(sgnum, D)]...)...]
+
+
+    wps=get_wycks(sgnum, D)
+    sg=spacegroup(sgnum, D)
+
+    wps_special=[]
+    sitesyms=[]
+    bandreps_array=Array{BandRep, 1}(undef, 0)
+
+    vec_index=1
+
+    first_k=irlabs[1][1]
+
+    dim_array=[i[1]==first_k for i in irlabs]
+
+    dim_array=dim_array.*irdims
+    
+    sitegroups=SiteGroup.(Ref(sg), wps)
+
+    maximal_wyckoffs=Array{WyckPos{2}, 1}(undef, 0)
+
+    for maximal_sitegroup in findmaximal(sitegroups)
+        push!(maximal_wyckoffs, maximal_sitegroup.wp)
+    end
+
+    for wp in maximal_wyckoffs
+        sitegroup=SiteGroup(spacegroup(sgnum, D), wp)
+
+        sitesym=_find_isomorphic_parent_pointgroup(sitegroup).label
+
+            sitegroup_symmetry_irreps=site_symmetry_irreps(sitegroup)
+            ρ_max=size(sitegroup_symmetry_irreps)[1]
+            for ρ in 1:ρ_max
+                mulliken_label=mulliken(sitegroup_symmetry_irreps[ρ])
+
+               # print(mulliken_label)
+                dim=dot(irvec2d[:, vec_index], dim_array)
+                 band_rep=BandRep(string( wp.mult, wp.letter), sitesym, string( wp.mult, wp.letter) , dim, false, false, irvec2d[:, vec_index], irlabs)
+                 push!(wps_special, string( wp.mult))
+                 push!(sitesyms, sitesym)
+                 push!(bandreps_array, band_rep )
+                 #print(band_rep)
+                 vec_index+=1
+        end
+    end
+    #bandreps=Array{BandRep, 1}(undef, 0)
+   # print(bandreps_array)
+
+    return BandRepSet(sgnum, bandreps_array, kvs, klabs, irlabs, allpaths, spinful, timeinvar)
+
+end
+
+BRS_VEC=[];
+
+BRS_NUMS=[];
+
+for br_num in 1:17
+    if br_num ∉[4, 7, 8, 12]
+        push!(BRS_VEC, make_bandrep_set(br_num, D=2))
+        push!(BRS_NUMS, br_num )
+    end
+end
```

Comment:
```suggestion
BRS_VEC = make_bandrep_set.(1:17, D=2)
```
Let's include the nonsymmorphic plane groups as well. You can do without the `BRS_NUMS` as well.